core/state_transition.libevm_test.go

```diff
@@ -54,3 +60,142 @@ func TestCanExecuteTransaction(t *testing.T) {
 	_, err := core.ApplyMessage(evm, msg, new(core.GasPool).AddGas(30e6))
 	require.EqualError(t, err, makeErr(msg.From, msg.To, value).Error())
 }
+
+func TestMinimumGasConsumption(t *testing.T) {
+	// All transactions will be basic transfers so consume [params.TxGas] by
+	// default.
+	tests := []struct {
+		name             string
+		gasLimit, refund uint64
+		minConsumption   uint64
+		wantUsed         uint64
+	}{
+		{
+			name:           "consume_extra",
+			gasLimit:       1e6,
+			minConsumption: 5e5,
+			wantUsed:       5e5,
+		},
+		{
+			name:           "consume_extra",
+			gasLimit:       1e6,
+			minConsumption: 4e5,
+			wantUsed:       4e5,
+		},
+		{
+			name:           "no_extra_consumption",
+			gasLimit:       50_000,
+			minConsumption: params.TxGas - 1,
+			wantUsed:       params.TxGas,
+		},
+		{
+			name:           "zero_min",
+			gasLimit:       50_000,
+			minConsumption: 0,
+			wantUsed:       params.TxGas,
+		},
+		{
+			name:           "consume_extra_by_one",
+			gasLimit:       1e6,
+			minConsumption: params.TxGas + 1,
+			wantUsed:       params.TxGas + 1,
+		},
+		{
+			name:           "min_capped_at_limit",
+			gasLimit:       1e6,
+			minConsumption: 2e6,
+			wantUsed:       1e6,
+		},
+		{
+			// Although this doesn't test minimum consumption, it demonstrates
+			// the expected outcome for comparison with the next test.
+			name:     "refund_without_min_consumption",
+			gasLimit: 1e6,
+			refund:   1,
+			wantUsed: params.TxGas - 1,
+		},
+		{
+			name:           "refund_with_min_consumption",
+			gasLimit:       1e6,
+			refund:         1,
+			minConsumption: params.TxGas,
+			wantUsed:       params.TxGas,
+		},
+	}
+
+	// Very low gas price so we can calculate the expected balance in a uint64,
+	// but not 1 otherwise tests would pass without multiplying extra
+	// consumption by the price.
+	const gasPrice = 3
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			hooks := &hookstest.Stub{
+				MinimumGasConsumptionFn: func(limit uint64) uint64 {
+					require.Equal(t, limit, tt.gasLimit)
```

Comment:
The arguments are `expected, actual`
```suggestion
					require.Equal(t, tt.gasLimit, limit)
```